packages/chain-adapters/src/utils/validateAddress.ts

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios'
+
+const TRM_LABS_API_URL = 'https://api.trmlabs.com/public/v1/sanctions/screening'
+
+const cache: Record<string, Promise<boolean>> = {}
+
+const _validateAddress = async (address: string): Promise<boolean> => {
+  type trmResponse = [
+    {
+      address: string
+      isSanctioned: boolean
+    },
+  ]
+
+  const response = await axios.post<trmResponse>(
+    TRM_LABS_API_URL,
+    [
+      {
+        address,
+      },
+    ],
+    {
+      headers: {
+        Accept: 'application/json',
+      },
+    },
+  )
+
+  return response.data[0].isSanctioned
+}
+
+export const validateAddress = async (address: string): Promise<void> => {
+  // dedupe and cache promises in memory
+  if (cache[address] === undefined) {
+    const newEntry = _validateAddress(address)
```

Comment:
we need to cache the response for this to be useful

Reply:
so looks like this does actually work due to the fulfilled promise reference. I feel like it would be much more readable to just cache the response imo.

Reply:
yeah we cache the promise to avoid stacking up duplicate calls async. Less readable for sure, but absolutely more practical with react spamming this